source/Ship.cpp

```diff
@@ -479,12 +482,13 @@ void Ship::FinishLoading(bool isNewInstance)
 	// definition stored safely in the ship model, which will not be destroyed
 	// until GameData is when the program quits. Also copy other attributes of
 	// the base model if no overrides were given.
-	if(GameData::Ships().Has(modelName))
+	if(GameData::Ships().Has(trueName))
 	{
-		const Ship *model = GameData::Ships().Get(modelName);
+		const Ship *model = GameData::Ships().Get(trueName);
 		explosionWeapon = &model->BaseAttributes();
-		if(pluralModelName.empty())
-			pluralModelName = model->pluralModelName;
+		// Allow for renames of player owned ships' models from game data.
+		modelName = model->ModelName();
+		pluralModelName = model->pluralModelName;
```

Comment:
If this is just for player-owned ships, should there not be an `IsYours` / government check guarding it?